packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -821,7 +821,8 @@ public void run() {
                   : MetadataChanges.EXCLUDE;
           listenerRegistrations.put(
               handle,
-              getDocumentReference(arguments).addSnapshotListener(metadataChanges, observer));
+              getDocumentReference(arguments)
+                  .addSnapshotListener(activity, metadataChanges, observer));
```

Comment:
Same as above `activity` will be `null` when the App is run in background only.
It really should be annotated with `@Nullable`.